Multilabel Docs Update (#3523)
apache/tinkerpop: docs/src/upgrade/release-4.x.x.asciidoc

Until now, vertices in the property graph model were limited to a single, immutable label assigned at creation. This
release introduces configurable label cardinality, allowing vertices to carry multiple labels that can be added and
removed over their lifetime. This is an optional feature for graph providers and not all will support it in the same
way. TinkerGraph enables it through a graph-level `LabelCardinality` setting that defaults to `ONE`, preserving the
existing single-label behavior:

```

> gremlin.tinkergraph.vertexLabelCardinality=ZERO_OR_MORE

Reviewer: ```suggestion
gremlin.tinkergraph.vertexLabelCardinality=ONE
```

The three modes are `ONE` (single, immutable, the 3.x default), `ONE_OR_MORE` (mutable, minimum one), and
`ZERO_OR_MORE` (fully flexible). Edge labels remain fixed at `ONE`.

With multi-label enabled, `addLabel()` and `dropLabel()` mutate a vertex's labels, and the new `labels()` step

> reliably returns every label — unlike `label()`, which returns only one and is now deprecated:

Reviewer: ```suggestion
which produces all labels, while the legacy `label()` step continues to produce a single label only, and is now deprecated:
```

> ```text

Reviewer: This should be using Java or groovy syntax highlighting.

==>manager
```

`mergeV()` accepts a list for `T.label` to match or create multi-label vertices, with append-only `onMatch`
semantics. By default `elementMap()` and `valueMap()` still return a single label string; use `with("multilabel")`,
per-traversal or as a persistent source, to receive the full set. The reference documentation covers each of these

> behaviors in detail.

Reviewer: ```suggestion
`mergeV()` now optionally accepts a list for `T.label` to match or create multi-label vertices. By default
`elementMap()` and `valueMap()` still return a single label string and a new traversal-source level
`with("multilabel")` config alters the results to wrap labels in lists. The reference documentation covers each of these
behaviors in detail.
```

===== Multi-Label Support

> Providers declare their supported label cardinality via `Graph.Features.VertexFeatures#getLabelCardinality()`, which

Reviewer: Could use more sensitivity around the concept of "default".
